q2_fmt/_peds.py

```diff
@@ -466,11 +479,458 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
     donor_df = donor_df.to_numpy()
     donor_mask = donor_df[donor_index_masking]
     donor_mask = donor_mask.astype(int)
-    recipdf = recip_df.to_numpy()
-    recipdf = recipdf.astype(int)
     return donor_mask
 
 
 def _mask_recipient(donor_mask, recip_df):
-    maskedrecip = donor_mask & recip_df
-    return maskedrecip
+    recip_mask = donor_mask & recip_df
+    return recip_mask
+
+
+def peds_simulation(table: pd.DataFrame, metadata: qiime2.Metadata,
+                    time_column: str, reference_column: str,
+                    subject_column: str,
+                    filter_missing_references: bool = False,
+                    drop_incomplete_subjects: bool = False,
+                    drop_incomplete_timepoint: list = None,
+                    iterations: int = 999) -> (pd.DataFrame, pd.DataFrame):
+
+    metadata_df = metadata.to_dataframe()
+    reference_series = _check_reference_column(metadata_df,
+                                               reference_column)
+
+    (metadata_df,
+     used_references) = _filter_associated_reference(reference_series,
+                                                     metadata_df, time_column,
+                                                     filter_missing_references,
+                                                     reference_column)
+
+    if len(used_references.unique()) == 1:
+        raise AssertionError("There is only one donated microbiome in your"
+                             " data. A Monte Carlo simulation shuffles"
+                             " your donated microbiome and recipient pairing"
+                             " and needs more than one donated microbiome"
+                             " to successfully shuffle.")
+    recipient = metadata_df.loc[metadata_df[reference_column].notnull()]
+
+    if len(recipient.index.unique()) == 1:
+        raise AssertionError("There is only one recipient in your"
+                             " data. A Monte Carlo simulation shuffles"
+                             " your donated microbiome and recipient pairing"
+                             " and needs more than one recipient"
+                             " to successfully shuffle.")
+
+    peds = sample_peds(
+           table=table, metadata=metadata,
+           time_column=time_column,
+           reference_column=reference_column,
+           subject_column=subject_column,
+           filter_missing_references=filter_missing_references,
+           drop_incomplete_subjects=drop_incomplete_subjects,
+           drop_incomplete_timepoint=drop_incomplete_timepoint).set_index("id")
+    actual_temp = peds["measure"]
```

Comment:
I would call this `actual_peds`, just to make the variable name more self-documenting.